src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -83,8 +84,14 @@ func (r *StatefulSetResource) shouldUsePartitionedUpdate(
 		return false, err
 	}
 
+	configuratorContainer, err := findContainer(sts.Spec.Template.Spec.InitContainers, configuratorContainerName)
```

Comment:
The relationships of when gets what updated is pretty complicated, I had to write this down.

```
I am running operator 21.1.1
It was started with --configurator-tag=v21.1.1

I deploy redpanda version 21.1.1
I want to upgrade redpanda from 21.1.1 to 21.2.0 so I update CRD with image tag 21.2.0
Now I am running redpanda 21.2.0 with operator 21.1.1 and it was configured using configurator 21.1.1

Now I update operator to 21.3.0
I am still running redpanda 21.2.0 managed by that operator
Update of the operator triggers restart of all redpanda nodes and the configurator will get updated to 21.3.0 still running redpanda 21.2.0
```

I don't like the second restart of the cluster to be honest. do you think it's necessary?

Reply:
I can see that this might be needed when the configurator actually fixes some bug during initialization but do we feel that rolling restart of already deployed redpanda that's running makes sense because of operator deploy (and thus configurator change?). Really just question :) I don't know the right answer

Reply:
Right, I don't think we should be triggering a restart in order to update an init container image.. At the same time, not being able to update the init image is also not desired. That is the current state.

A possible solution would be to associate the configurator version with the cluster CR (be equal to Redpanda's tag or specify the configurator tag explicitly in the CR) and have both image updates happen simultaneously if that is desired (or separately). What do you think?

Reply:
It would be trivial to update this PR to work with the above proposal.

Reply:
What do you think about this: 

```
Now I update operator to 21.3.0
I am still running redpanda 21.2.0 managed by that operator
Update of the operator DOES NOT trigger restart of all redpanda nodes configurator won't get updated

Someone triggers update of redpanda to 21.2.2
Configurator gets updated to 21.3.0 and that configurator starts redpanda 21.2.2
```

I think that should be doable only with a tweak of `shouldUsePartitionedUpdate` function, perhaps?